Sources/Imaging/Core/AbstractImageInterpolator/index.js

```diff
@@ -0,0 +1,240 @@
+        dataTypeSize * c
+      );
+      iinfo.numberOfComponents = 1;
+
+      const v = [value];
+      publicAPI.interpolatePoint(iinfo, p, v);
+      value = v[0];
+    }
+    return value;
+  };
+
+  publicAPI.interpolate = (point, value) => {
+    const p = [
+      (point[0] - model.origin[0]) / model.spacing[0],
+      (point[1] - model.origin[1]) / model.spacing[1],
+      (point[2] - model.origin[2]) / model.spacing[2],
+    ];
+
+    if (publicAPI.checkBoundsIJK(p)) {
+      publicAPI.interpolatePoint(model.interpolationInfo, p, value);
+      return true;
+    }
+
+    for (let i = 0; i < model.interpolationInfo.numberOfComponents; ++i) {
+      value[i] = model.outValue;
+    }
+    return false;
+  };
+
+  publicAPI.computeNumberOfComponents = (inputCount) => {
+    const component = Math.min(
+      Math.max(model.componentOffset, 0),
+      inputCount - 1
+    );
+    const count =
+      model.componentCount < inputCount - component
+        ? model.componentCount
+        : inputCount - component;
+    return count > 0 ? count : inputCount - component;
+  };
+
+  publicAPI.getNumberOfComponents = () =>
+    model.interpolationInfo.numberOfComponents;
+
+  publicAPI.interpolateIJK = (point, value) => {
+    publicAPI.interpolatePoint(model.interpolationInfo, point, value);
+  };
+
+  publicAPI.checkBoundsIJK = (point) =>
+    !(
+      point[0] < model.structuredBounds[0] ||
+      point[0] > model.structuredBounds[1] ||
+      point[1] < model.structuredBounds[2] ||
+      point[1] > model.structuredBounds[3] ||
+      point[2] < model.structuredBounds[4] ||
+      point[2] > model.structuredBounds[5]
+    );
+
+  publicAPI.computeSupportSize = null; // (matrix) => {};
+  publicAPI.isSeparable = null;
+  publicAPI.precomputeWeightsForExtent = (matrix, inExtent, checkExtent) => {};
+  publicAPI.FreePrecomputedWeights = (weights) => {
+    /*
+    for (let k = 0; k < 3; ++k) {
+      const step = weights.kernelSize[k];
+      // TODO: check if ok
+      weights.positions[k] += step * weights.weightExtent[2 * k];
+      if (weights.weights[k]) {
+        // TODO: check if ok
+        weights.weights[k] += step * weights.weightExtent[2 * k];
+      }
+    }
+
+    if (weights.workspace) {
+      for (let i = 1; i < weights.kernelSize[1]; ++i) {
+        // TODO
+        ...
+      }
+    }
+    */
+  };
+  publicAPI.interpolatePoint = (interpolationInfo, point, value) => {};
+  publicAPI.interpolateRow = (weights, xIdx, yIdx, zIdx, value, n) => {};
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+const DEFAULT_VALUES = {
+  outValue: 0,
+  tolerance: Number.EPSILON,
+  componentOffset: 0,
+  componentCount: -1,
+  borderMode: ImageBorderMode.CLAMP,
+  slidingWindow: false,
+
+  scalars: null,
+  interpolationInfo: Object.assign({}, vtkInterpolationInfo),
```

Comment:
I'm guessing that you know that all the arrays inside are just reference to a single instance of those array?

Reply:
Yes, I use `subarray` as an address pointer. I don't want to copy the array only to point at different places in the array.

Reply:
subarray will recreate a new array...

Reply:
It's a new array but it points on the same array buffer. So it is fast (no copy)

Reply:
Ok I was talking about the array you define by [] inside vtkInterpolationInfo.